addon/mixins/service-mixin.js

```diff
@@ -70,7 +70,11 @@ export default Ember.Mixin.create({
 
     // Ember.merge(form, this.get('defaultParams'));
     return Object.keys(form).map((key) => {
-      return [key, form[key]].map(encodeURIComponent).join('=');
+      // don't convert null props into fields
+      // https://github.com/emberjs/ember.js/blob/3b60016c2b6b38646d954014d10f74fcfe1a3d54/packages/ember-runtime/lib/core.js#L82-84
+      if (!Ember.isNone(form[key])) {
+        return [key, form[key]].map(encodeURIComponent).join('=');
+      }
```

Comment:
Won't this give us `undefined` instead of null in the array? If it doesn't fall into the `if` it will return `undefined` I think.

Reply:
Good catch @mjuniper, there will be `undefined` elements in the `.map()` output, and then they get sent to `.join()` and you end up w/ additional `&` in the final string.

![image](https://user-images.githubusercontent.com/662944/32969555-6d231a60-cb9a-11e7-8f72-440f87c87cfc.png)

It would be better to build up a new array using a `.forEach()` instead of `.map()` here.

Reply:
Or we could use `reduce`. Something like:

```
return Object.keys(form).reduce((acc, key) => {
  if (!Ember.isNone(form[key])) {
    acc.push([key, form[key]].map(encodeURIComponent).join('='));
  }
  return acc;
}, []).join('&');
```

Reply:
updating to that! much better! jupe++